File: Assignment no. 71/Q1.java
import java.io.*;

class Rotation
{
	public boolean chkRotation(String str1,String str2)
	{
		char[] arr=str1.toCharArray();
		char[] brr=str2.toCharArray();
		int i=0,j=0;
		
		while((j<brr.length)&&(arr[0]!=brr[j]))
		{
			j++;
		}
		if(j==brr.length)
		{
			return false;
		}
		for(;(i<arr.length)&&(arr[i]==brr[j%(brr.length)]);i++,j++);
		
		if(i<arr.length)
		{
			return false;
		}
		else
		{
			return true;
		}
	}
}

class Q1
{
	public static void main(String arg[]) throws Exception
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		System.out.println("Enter first string :");
		String str1=br.readLine();
		System.out.println("Enter second string :");
		String str2=br.readLine();
		Rotation obj=new Rotation();
		if(obj.chkRotation(str1,str2))
		{
			System.out.println("String is rotation string");
		}
		else
		{
			System.out.println("String is not a rotation string");
		}
	}
}

File: Assignment no. 39/Q1.c
#include<stdio.h>
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//  Name        :   strlen
//  Input       :   character address,character address
//  Output      :   none
//  Description :   display length of two strings
//  Author      :   Nikhil
//  Date        :   26/10/2019
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

void strlen(char *str1,char *str2)
{
    int iCnt1=0,iCnt2=0;
    if((str1==NULL)||(str2==NULL))
    {
        return;
    }
    while((*str1!='\0')||(*str2!='\0'))
    {
        if(*str1!='\0')
        {
            iCnt1++;
            str1++;
        }
        if(*str2!='\0')
        {
            iCnt2++;
            str2++;
        }
    }
    printf("Length of first string is :\n%d",iCnt1);
    printf("\nLength of second string is :\n%d",iCnt2);
}

//Entry point function : main()
int main()
{
    char Arr[30]={'\0'};
    char Brr[30]={'\0'};
    printf("Enter first string :\n");
    scanf("%[^'\n']s",Arr);
    fflush(stdin);
    printf("Enter second string :\n");
    scanf("%[^'\n']s",Brr);
    strlen(Arr,Brr);
    return 0;                       // return to operating system
}

